lib/install/management/create_test.go

```diff
@@ -245,10 +249,27 @@ func testCreateAppliance(op trace.Operation, sess *session.Session, conf *config
 		} else {
 			t.Fatal(err)
 		}
-		return
+		return nil
 	}
 
 	if hasErr {
 		t.Errorf("No error when error is expected.")
 	}
+
+	return d.appliance
+}
+
+func testRelocateAppliance(op trace.Operation, appliance *vm.VirtualMachine, sess *session.Session, conf *config.VirtualContainerHostConfigSpec, vConf *data.InstallerData, t *testing.T) {
+	d := &Dispatcher{
+		session:   sess,
+		op:        op,
+		isVC:      sess.IsVC(),
+		appliance: appliance,
+		force:     false,
+	}
+
+	err := d.relocateAppliance()
```

Comment:
There are multiple early exist conditions in relocateAppliance() function. How do you plan to cover them in unit tests?

Reply:
A lot of the details in `relocateAppliance()` are pending and will be iterated upon. For example, this changeset uses random host placement since ranked placement is still being worked on. There's also some discussion to move some of the placement/policy logic deeper into the VM methods, in which case `relocateAppliance()` wouldn't have to worry about them at all.

Reply:
Yeah pretty soon most of this code will be unneeded as all the related logic will be abstracted away under `vm.VirtualMachine.PowerOn`.